api/app/signals/apps/signals/managers.py

```diff
@@ -280,12 +291,19 @@ def _update_location_no_transaction(self, data, signal):
         :returns: Location object
         """
         from .models import Location
-        from .utils.location import _get_stadsdeel_code
+        from .utils.location import _get_stadsdeel_code, _get_area
 
         # SIG-2513 Determine the stadsdeel
         default_stadsdeel = data['stadsdeel'] if 'stadsdeel' in data else None
         data['stadsdeel'] = _get_stadsdeel_code(data['geometrie'], default_stadsdeel)
 
+        # set area_type and area_code if default area type is provided
+        if DEFAULT_SIGNAL_AREA_TYPE:
```

Comment:
We need a test here to check that these fields are correctly updated when the location is updated.